lib/launcher.js

```diff
@@ -54,6 +55,13 @@ class Launcher {
         await fs.writeFile(this.files.packageJSON, JSON.stringify(packageData, ' ', 2))
         await fs.rm(path.join(this.projectDir, '.config.nodes.json'), { force: true })
         await fs.rm(path.join(this.projectDir, '.config.nodes.json.backup'), { force: true })
+
+        // as the package.json file has been re-written, we will "assume" different versions
+        // of the modules and therefore remove the package-lock.json file to permit modules
+        // to be updated to the versions specified in package.json
+        if (existsSync(this.files.packageLockJSON)) {
+            await fs.rm(this.files.packageLockJSON)
```

Comment:
Wouldn't an `npm install` already handle this? 

Or are you wanting to run `npm upgrade` to upgrade the package log speficially?

Reply:
Pez, was discussed in slack

> 
> Steve
> When switching snapshots with different versions of Node-RED specified, "Installing Packages" is ran but Node-RED version remains old.
> npm install --production (as the launcher and agent execute) ignore package.json and respect the lock file instead
> Should we be inspecting the package or indiscriminately deleting the lock file in the agent device project dir when "Updating configuration files"-> writePackage() is called?
> 2 replies
> 
> 
> nick
> My instinct is to delete the package lock file whenever we update the package file
> 
> 
> Steve

> same - but wanted to check

Reply:
To me, this could lead to unexpected upgrades outside of the package changes, but so would running an explicit npm upgrade, so unblocking until we find a better solution for now.

Reply:
On second thought, this still doesn't fully grok for me actually, an `npm install` always updates the package-lock based on the latest package.json as part of the install. Testing locally.

Reply:
Result of my tests in https://github.com/FlowFuse/device-agent/issues/178#issuecomment-1750306065